src/plumpy/processes.py

```diff
@@ -1001,13 +1056,19 @@ def close(self) -> None:
     # region State related methods
 
     def transition_failed(
-        self, initial_state: Hashable, final_state: Hashable, exception: Exception, trace: TracebackType
+        self,
+        initial_state: Hashable,
+        final_state: Hashable,
+        exception: Exception,
+        trace: TracebackType,
     ) -> None:
         # If we are creating, then reraise instead of failing.
         if final_state == process_states.ProcessState.CREATED:
             raise exception.with_traceback(trace)
 
-        self.transition_to(process_states.ProcessState.EXCEPTED, exception, trace)
+        state_class = self.get_states_map()[process_states.ProcessState.EXCEPTED]
+        new_state = self._create_state_instance(state_class, exception=exception, trace_back=trace)
```

Comment:
This is a good point, I did this for future factoring after and in the end I arrive at only accept the `ProcessState` enum type as argument instead of accepting state object, you can have a look at https://github.com/unkcpz/plumpy/blob/6bfb87df69889f0082e8ecbe7732d24ad69e64c2/src/plumpy/base/state_machine.py#L159-L164

Is that more clear? I was plan to include change for a later release and am working on more refactoring for some interfaces.

Reply:
Okay, I do the other way as you suggested, passing the `process_states.ProcessState.EXCEPTED` directly to the `_create_state_instance` and having `self.get_states_map` call from inside.